docs/user-guide/connecting-applications.md

```diff
@@ -102,24 +110,36 @@ spec:
   selector:
     app: nginx
 ```
-This specification will create a Service which targets TCP port 80 on any Pod with the "app=nginx" label, and expose it on a targetPort (set to the `port` by default) that other pods can use to access the Service. Check your Service:
+This specification will create a Service which targets TCP port 80 on any Pod with the `app=nginx` label, and expose it on an abstracted Service port (`targetPort`: is the port the container accepts traffic on, `port`: is the abstracted Service port, which can be any port other pods use to access the Service). Check your Service:
 ```shell
 $ kubectl get svc
-NAME         LABELS     SELECTOR    IP(S)          PORT(S)
-nginxsvc     <none>     app=nginx   10.0.116.146   80/TCP
+NAME         LABELS        SELECTOR    IP(S)          PORT(S)
+nginxsvc     app=nginx     app=nginx   10.0.116.146   80/TCP
 ```
 
 As mentioned previously, a Service is backed by a group of pods. These pods are exposed through `endpoints`. The Service's selector will be evaluated continuously and the results will be POSTed to an Endpoints object also named `nginxsvc`. When a pod dies, it is automatically removed from the endpoints, and new pods matching the Service’s selector will automatically get added to the endpoints. Check the endpoints, and note that the ips are the same as the pods created in the first step:
 ```shell
+$ kubectl describe svc nginxsvc
+Name:			nginxsvc
+Namespace:		default
+Labels:			app=nginx
+Selector:		app=nginx
+Type:			ClusterIP
+IP:			10.0.116.146
+Port:			<unnamed>	80/TCP
+Endpoints:		10.245.0.14:80,10.245.0.15:80
+Session Affinity:	None
+No events.
+
 $ kubectl get ep
 NAME         ENDPOINTS
 nginxsvc     10.245.0.14:80,10.245.0.15:80
 ```
-You should now be able to curl the nginx Service on `10.0.208.159:80` from any node in your cluster. Note that the Service ip is completely virtual, it never hits the wire, if you’re curious about how this works you can read more about the [service proxy](services.md#virtual-ips-and-service-proxies).
+You should now be able to curl the nginx Service on `10.0.116.146:80` from any node in your cluster. Note that the Service ip is completely virtual, it never hits the wire, if you’re curious about how this works you can read more about the [service proxy](services.md#virtual-ips-and-service-proxies).
 
-## Accessing the Service from other pods in the cluster
+## Accessing the Service
 
-Kubernetes supports 2 primary modes of finding a Service - environment variables and DNS. The former works out of the box while the latter requires the [kube-dns cluster addon](../../cluster/addons/dns/README.md).
+Kubernetes supports 2 primary modes of service discovery - environment variables and DNS. The former works out of the box while the latter requires the [kube-dns cluster addon](../../cluster/addons/dns/README.md).
```

Comment:
Technically, the previous text was more correct. Discovery is about detection of which services exist, such as pulling up a list of available printers on your network. We just do name resolution.